File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        
        finalnums = []
        nums.sort() 
        
        prev = 1
       
        for x, a in enumerate(nums):
            
            if x > 0 and a == nums[x - 1]:
                continue
                
            
            firstNum = nums[x]
            
            left = x+1
            right = len(nums) -1
            
            
 
            while left < right:
        
                total = nums[right] + nums[left] + firstNum
                  
                if  total < 0:
                    left += 1
                elif total > 0:
                    right -= 1
                else:
                    finalnums.append([firstNum,nums[left],nums[right]])
                    left+=1
                    while(nums[left] == nums[left-1] and left < right):
                        left += 1 
            prev = firstNum
                
        return finalnums
                    
        
                
                
        
        
        

File: 0015-3sum/0015-3sum.java
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        
        nums.sort()
        p = []

        for i, a in enumerate(nums):

            if(i > 0 and a == nums[i-1]):
                continue


            y = i + 1
            z = len(nums) - 1

            while(y < z):
            
                sum1 = a + nums[y] + nums[z]

                if(sum1 > 0):
                    z-=1
                
                elif(sum1 < 0):
                    y+=1

                else:
                    p.append([a, nums[y], nums[z]] )
                    y+=1
                    while(nums[y] == nums[y-1] and y < z):
                        y+=1
                    

        return p




        